### PR TITLE
Miawong/sc 55403/improve the gitops ux

### PR DESCRIPTION
**Pull request by miaawong**
#### What this PR does / why we need it:
UI updates to the GitOps workflow to improve ease of use. 
https://www.loom.com/share/0a48eae6e3834b53a479b64f1598b3f2

#### Which issue(s) this PR fixes:

Fixes # https://app.shortcut.com/replicated/story/55403/improve-the-gitops-ux

#### Does this PR introduce a user-facing change?

```release-note
UI updates to the GitOps workflow to improve ease of use. 
```

#### Does this PR require documentation?

** will add later

### CODE REVIEW

**Review by miaawong — COMMENTED** [review]
everything is a mess right now, i will have a PR that will refactor this feature down and align with the bulletproof structure

File: web/src/config-ui/subNavConfig.js [hunk 1]
```diff
@@ -67,12 +67,6 @@ export default [
       Boolean(app.name) &&
       Utilities.sessionRolesHasOneOf([rbacRoles.CLUSTER_ADMIN]),
   },
-  {
```
Comment (miaawong) [inline web/src/config-ui/subNavConfig.js:70]: remove gitops from subnav bar

File: web/src/components/gitops/SetupProvider.jsx [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+              >
+                {disablingGitOps
+                  ? "Disabling GitOps"
+                  : "Disable GitOps for this app"}
+              </a>
+            )}
+          </div>
+        </div>
+        {renderGitOpsProviderSelector({
+          owner,
```
Comment (miaawong) [inline web/src/components/gitops/SetupProvider.jsx:244]: I initially thought I could take everything out and put it into this component but it turns out that alot of data is being shared. so this renderGitOpsProviderSelector is actually rendered in the parent component. I will be using Context API to make the code better in the next PR.

File: web/src/components/gitops/GitOpsDeploymentManager.jsx [hunk 3]
```diff
@@ -369,11 +452,11 @@ class GitOpsDeploymentManager extends React.Component {
...
-    if (this.isSingleApp()) {
-      this.stepFrom("provider", "action");
-    } else {
-      this.finishSetup();
-    }
+    //  if (this.isSingleApp()) {
+    this.stepFrom("provider", "action");
+    // } else {
+    //   this.finishSetup();
+    // }
```
Comment (rianfowler) [inline web/src/components/gitops/GitOpsDeploymentManager.jsx:459]:
```suggestion
    this.stepFrom("provider", "action");
```

File: web/src/components/gitops/GitOpsRepoDetails.jsx [hunk 4]
```diff
@@ -121,19 +144,36 @@ class GitOpsRepoDetails extends React.Component {
...
+
+    return true;
+    // const isAllowed =
+    //   owner !== this.props.owner ||
+    //   repo !== this.props.repo ||
+    //   branch !== this.props.branch ||
+    //   path !== this.props.path ||
+    //   action !== this.props.action ||
+    //   format !== this.props.format;
+    // return isAllowed;
```
Comment (rianfowler) [inline web/src/components/gitops/GitOpsRepoDetails.jsx:176]:
```suggestion

```

**Review by rianfowler — DISMISSED** [review]

**Review by rianfowler — APPROVED** [review]

**Pull request merged**